runtime/src/in_mem_accounts_index.rs

```diff
@@ -49,9 +49,18 @@ impl<T: IsCached> InMemAccountsIndex<T> {
         result
     }
 
-    pub fn items(&self) -> Vec<(K, AccountMapEntry<T>)> {
+    pub fn items<R>(&self, range: &Option<&R>) -> Vec<(K, AccountMapEntry<T>)>
+    where
+        R: RangeBounds<Pubkey> + std::fmt::Debug,
+    {
         Self::update_stat(&self.stats().items, 1);
-        self.map.iter().map(|(k, v)| (*k, v.clone())).collect()
+        let mut result = Vec::with_capacity(self.map.len());
```

Comment:
If the range is `None`, I agree the capacity parameter should be `map.len()`. If the range is `Some`, would it be beneficial to set the capacity to the range's width instead?

(This is me assuming that keys are unique, so if they are not, then the current impl makes sense.)

Reply:
range width is specified in pubkey values. We don't know how many pubkeys are in a range width ex: (0xffde... to 0xffdf...). We only know that we will return <= map.len, which is all the entries present in the accounts index. I don't think we can even get the width of a range of pubkeys. Perhaps as a 32 byte integer value, which is bigger than usize at any rate. I think map.len is the only thing that makes sense.

Reply:
Ah, yeah, that makes sense. I was thinking small scale and none-sparse. Thanks for spelling it out for me.